node/src/service.rs

```diff
@@ -62,6 +65,29 @@ pub type Service<RuntimeApi> = PartialComponents<
     (ParachainBlockImport<RuntimeApi>, Option<Telemetry>, Option<TelemetryWorkerHandle>),
 >;
 
+pub fn create_extrinsic_filter(
+    enable_filter: bool,
+    log_rejections: bool,
+) -> Arc<impl sp_avn_common::transaction_filter::ExtrinsicFilter> {
+    use codec::Decode;
+    use avn_parachain_runtime::{RuntimeCall, UncheckedExtrinsic};
+
+    let decoder = |bytes: &[u8]| -> Result<RuntimeCall, codec::Error> {
+        UncheckedExtrinsic::decode(&mut &bytes[..]).map(|xt| xt.function)
+    };
+
+    let predicate = |call: &RuntimeCall| {
+        matches!(
+            call,
+            RuntimeCall::Balances(pallet_balances::Call::transfer_allow_death { .. }) |
+                RuntimeCall::Balances(pallet_balances::Call::transfer_keep_alive { .. }) |
+                RuntimeCall::Balances(pallet_balances::Call::transfer_all { .. })
+        )
+    };
```

Comment:
Would it be better to move this logic in the runtime (accessible via `avn_parachain`runtime`)? This way we could have a clear separation between the node code and the runtime (that is always owned by clients and can be changed easily).